sdk/storage/Azure.Storage.DataMovement/src/ChannelProcessing.cs

```diff
@@ -46,12 +46,13 @@ public static IProcessor<T> NewProcessor<T>(int readers, int? capacity = null)
             : new ParallelChannelProcessor<T>(channel, readers);
     }
 
-    private abstract class ChannelProcessor<TItem> : IProcessor<TItem>, IDisposable
+    private abstract class ChannelProcessor<TItem> : IProcessor<TItem>, IAsyncDisposable
     {
         /// <summary>
         /// Async channel reader task. Loops for lifetime of object.
         /// </summary>
         private Task _processorTask;
+        internal TaskCompletionSource<bool> _processerTaskCompletionSource;
```

Comment:
`private`?

Reply:
Actually realized this has to stay internal so it can be accessed by the Sequential and Parallel ChannelProcessors to inherit from it.